### DISCUSSION TITLE
Chyba při zadání API key

### DISCUSSION THREAD

**Discussion by Jancas86** [dvejsada/PID_integration#19] [Help needed!]
Zdravím, 
po stažení nové verze jsem se dostal k dialogovému oknu nastavení integrace, ale ani na několik pokusů a s dvěma novými klíči API z Golemio se nechce posunout dále a vyhodí chybu "Unknown API connection error".

Je taky možné, že zadávám zastávku ve špatném formátu. Zkoušel jsem přidat celý řádek ze stops_list.txt i různé části názvu zastávek, ale bez úspěchu.
Log je níže.

Tato chyba pochází z vlastní integrace.

Logger: custom_components.pid_integration.config_flow
Source: custom_components/pid_integration/config_flow.py:60
Integration: PID Departure Boards (documentation, issues)
First occurred: 16:55:53 (7 occurrences)
Last logged: 17:23:23

Unknown API connection error
Traceback (most recent call last):
  File "/config/custom_components/pid_integration/config_flow.py", line 60, in async_step_user
    info = await validate_input(self.hass, user_input)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/pid_integration/config_flow.py", line 30, in validate_input
    raise CannotConnect
custom_components.pid_integration.config_flow.CannotConnect

Druhý log s nenalezenou zastávkou:

Tato chyba pochází z vlastní integrace.

Logger: custom_components.pid_integration.config_flow
Source: custom_components/pid_integration/config_flow.py:60
Integration: pid_integration (documentation, issues)
First occurred: 16:54:32 (1 occurrences)
Last logged: 16:54:32

Stop with provided awsIDs was not found.
Traceback (most recent call last):
  File "/config/custom_components/pid_integration/config_flow.py", line 60, in async_step_user
    info = await validate_input(self.hass, user_input)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/pid_integration/config_flow.py", line 28, in validate_input
    raise StopNotFound
custom_components.pid_integration.config_flow.StopNotFound

![PID](https://github.com/dvejsada/PID_integration/assets/154517195/3a2dda1d-aa71-4e75-a438-a2ab48e236b2)

Díky za kontrolu a sdělení, co jsem zase udělal špatně:)
R

**Comment by dvejsada** [comment]
Co přesně jste zadával do prvního pole aswIds? Mělo by to vypadat např. 235_1

**Comment by dvejsada** [comment]
Bohužel se mi Vaši chybu nedaří nijak zreplikovat. Nicméně API kod mate správně, jinak by to hodilo jinou chybu. Podívejte se tedy jestě jednou, jestli zadáváte tu zastávku dle komentáře výše a dle dokumenetace.

**Comment by Jancas86** [comment]
Moje chyby je těžké opakovat🙂 Samozřejmě jsem zadával špatně zastávku. Díky za úpravu v popisu. Teď už jen přijít na to, jak zobrazit čas odjezdu. Předpokládám, že musím přinutit HA někde zobrazit atribut  Arrival timestamp. Už začínám chápat co znamená věnovat tomu čas🙂. Díky za to, co děláte. Máte u mě kafe.

**Reply by dvejsada** [reply]
Bohužel, k přípravě nějaké custom karty na zobrazení na Dashboardu jsem se zatím nedostal. Z diskuse na facebooku využívali např. html card, markdown card či flex card.

**Reply by tomassvec** [reply]
Dovoluji si poskytnout svoji jednoduchou kartu v markdown, která mi dělá a zobrazuje to co chci.

type: markdown
content: >
  ||||||

  |:--|:--|:--|--:|:--|

  {% for state in states.sensor -%}

  {%- if state.name.startswith("Všenory  departure") %}{{
  as_datetime(state.attributes.arrival_timestamp.scheduled).strftime('%H:%M')
  }}&nbsp;&nbsp;|{{state.attributes.trip.short_name}}&nbsp;&nbsp;|{{state.attributes.trip.headsign}}|{%-
  if state.attributes.delay.is_available %}{%- if
  state.attributes.delay.minutes>0 %} +{{state.attributes.delay.minutes}}m{%
  else %}<b>(včas)</b>{% endif -%}{% endif -%}&nbsp;&nbsp;|{%- if
  state.attributes.last_stop.name!=None %}`Poloha:
  {{state.attributes.last_stop.name}}``{% endif -%}|

  {% endif -%}

  {%- endfor %}.
